app/views/pets/show.html.erb

```diff
@@ -7,4 +7,9 @@
 <p>Availability: <%= @pet.adoption_status %></p>
 <%= link_to "Update #{@pet.name}'s Profile", "/shelters/#{@pet.shelter.id}/pets/#{@pet.id}/edit" %> /
 <%= link_to "Delete #{@pet.name}", "/shelters/#{@pet.shelter.id}/pets/#{@pet.id}", method: :delete %> /
-<%= link_to "Add to Favorites", "/favorite/#{@pet.id}", method: :patch %>
\ No newline at end of file
+
+<% if favorites.pets.keys.include?(@pet.id.to_s) %>
```

Comment:
checks if the `pet_id` for the show view page has been added to the `favorites` before. If it has been added before, the link shown allows to remove the pet from the `favorites` instead.